### ISSUE TITLE
Importing Records With Existing Data Creates 'Invalid File Format' Error

### ISSUE THREAD

**Issue by chrisjlocke** [sqlitebrowser/sqlitebrowser#1590]
Details for the issue
--------------------
This semi-relates to an issue raised in #1585.  Upon further testing, a reproducible set of steps was identified.

When importing data into a table, if an imported record fails a constraint check (eg, primary key violation, eg, importing record #1 again) DB4S throws up a 'invalid file format' error, then a further 'the system is busy - do you want to cancel?' dialog.  If the user presses No, DB4S locks up*.

*DB4S didn't come out of the 'not responding state'

Tried the alpha, nightly, and the recent 'special build' for #1575.
See screencast: https://screencast-o-matic.com/watch/cF60IJYAsP

#### What did you do?
Imported records from a CSV file, which was from an exported table.  The import was into the same table, so records were known to violate the primary key.

#### What did you expect to see?
The records known to violate the primary key would be ignored.

#### What did you see instead?
DB4S got kicked in the nuts, fell down on its knees and didn't get back up again.

Useful extra information
-------------------------
Please see the below screencast:
https://screencast-o-matic.com/watch/cF60IJYAsP

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [x] Windows: ( _version:_ _10__ )
- [ ] Linux: ( _distro:_ ___ )
- [ ] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### What is your DB4S version?

- [x] 3.11.0-alpha*
- [ ] 3.10.1
- [x] Other: ___

#### Did you also

- [x] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [ ] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by MKleusberg** [comment]
Thanks Chris! That's a good find and a very helpful error description :+1: I think I have fixed both issues, wrong error message and crash. Can you check tomorrow's nightly?

I have also already cherry-picked this commit to the v3.11.x branch as it fixes an important regression which was introduced after 3.10.1 - it also doesn't change any user-visible strings, so no translation update needed :smile:

**Comment by chrisjlocke** [comment]
HAve just tried the latest nightly, and while there is a more 'valid' error:

![image](https://user-images.githubusercontent.com/7872218/47423184-52cec800-d77c-11e8-8ecd-8919392e4646.png)

the import terminates.

![image](https://user-images.githubusercontent.com/7872218/47423224-6c700f80-d77c-11e8-85d8-5fb0b24f402a.png)

Would it be possible to continue, or because we've received an error, and potentially that error could be anything, its 'safer' to terminate?
If the former, then would it be also possible to ignore certain errors?  

Theoretically, if I've deleted some records but have a table/database backup, I could import just the missing records.  I can't do that currently, as the import terminates at an error.  In an ideal world, DB4S could throw up a 'Retry, Ignore, Cancel' dialog...

![image](https://user-images.githubusercontent.com/7872218/47423705-9b3ab580-d77d-11e8-9684-6bb99d69f9dd.png)

This could give the user a chance to open the database in another instance to correct the error, and retry.

**Comment by MKleusberg** [comment]
Thanks for testing, Chris! You're quite right about the other point. I think your dialog would also need a 'Remember this action' checkbox or some similar option for larger CSV files. Maybe we also need to distinguish between different sorts of errors for that - but that is quite tricky because we only get a very vague error code from SQLite and the error string as seen in the message box. The first probably doesn't contain enough information on the error to be of any help here and the second is subject to change and hard to parse. I guess the easiest solution is to just end the import as we do now but add a dropdown box to allow changing the ON CONFLICT strategy, just as @antonybits suggested in #1585. The only downside is that you might have to start the import a second time if you didn't get the settings right on the first run. But since the import is so much faster now... :stuck_out_tongue_closed_eyes:
